### ISSUE TITLE
MrSID, ECW licenses may not be compatible by QGIS distribution

### ISSUE THREAD

**Issue by kyngchaos** [qgis/QGIS-Mac-Packager#15]
As far as I can tell for these, the distribution of the binary libraries within an application require the application to restrict download from certain countries, at least if distributed from the US.  I don't see any such restriction in the QGIS site.  This was the reason for my method requiring the user to download those libraries and manually install them for the GDAL format plugins (or maybe I was just being over cautious, I don't know).

This should be investigated and verified, and those libraries removed from the distribution if necessary.

**Comment by NathanW2** [comment]
Is this in the license file for the ECW SDK?

**Comment by kyngchaos** [comment]
ECW SDK and MrSID SDK

**Comment by nyalldawson** [comment]
> at least if distributed from the US

Is this actually the case here? No idea how mirrors may effect things, but I believe the QGIS hosting infrastructure is european based. @elpaso @rduivenvoorde ?

> This was the reason for my method requiring the user to download those libraries and manually install them for the GDAL format plugins

I'm curious how this approach gets around the issue -- is there something on your server backend blocking the downloads from the restricted regions?

**Comment by saberraz** [comment]
Another question is, how are we handling Windows packages, which come with those drivers already packaged?

**Comment by NathanW2** [comment]
They are bundled and would be very bad to remove them. If this is a US
trade issue like what gitlab and GitHub have recently have had to do, I am
not sure it will affect us as we are not a US company.

**Comment by kyngchaos** [comment]
> I'm curious how this approach gets around the issue -- is there something on your server backend blocking the downloads from the restricted regions?

No, I can't block downloads by region.  I only distribute the GDAL plugin binary that's linked to the SDK libraries.  It's up to the user to get the SDK libraries, I'm not distributing those.

**Comment by nyalldawson** [comment]
> It's up to the user to get the SDK libraries, I'm not distributing those.

Ah, makes sense!

**Comment by jef-n** [comment]
Hi Nyall,

On Mon, 29. Jul 2019 at 15:22:45 -0700, Nyall Dawson wrote:
> > This was the reason for my method requiring the user to download those
> > libraries and manually install them for the GDAL format plugins
 
> I'm curious how this approach gets around the issue -- is there something on
> your server backend blocking the downloads from the restricted regions? 

It also restricts distribution to "any national of these countries".


Jürgen

-- 
Jürgen E. Fischer           norBIT GmbH             Tel. +49-4931-918175-31
Dipl.-Inf. (FH)             Rheinstraße 13          Fax. +49-4931-918175-50
Software Engineer           D-26506 Norden            https://www.norbit.de
QGIS release manager (PSC)  Germany                    IRC: jef on FreeNode

norBIT Gesellschaft fuer Unternehmensberatung und Informationssysteme mbH
Rheinstrasse 13, 26506 Norden
GF: Juergen Fischer, Nils Kutscher HR: Amtsgericht Aurich HRB 100827
Datenschutzerklaerung: https://www.norbit.de/83/

**Comment by nyalldawson** [comment]
> It also restricts distribution to "any national of these countries".

(my unhelpful reply): what a steaming pile of horse 💩. This restriction is so antiquated and useless in the current era that it's just a laughable joke.

**Comment by NathanW2** [comment]
I also don't think it applies to us anyway given it has come up in other
open projects and the result is 'doesnt affect us'

**Comment by mbernasocchi** [comment]
FWIW we are also not distributing ECW and MrSID in QField for that reason

**Comment by mewoch** [comment]
does providing a build minus MrSID and ECW for those affected leave enough wriggle room?
